### PR TITLE
vitor redesign permissions page and user permission component

### PR DESCRIPTION
**Pull request by vitorsaa2k**
Decided to redesign the "Manage User Permissions" component and redesigned the permissions management page.

## Main changes
1. Better UX and UI for the Permissions Management page and to the "Manage User Permissions" component.
2. Now, there's a scrollbar if the length of roles is greater than the length of the base roles.

## Description
- The major of the changes is just about css and html. The only change on logic is on the addition of the useHistory hook the substitute the old <a>(Anchor) component.
- **You will need an Owner account the test this PR**
If you don't have one you can use this one: 
email: `email@owner.com`
password: `Vitor.01`

## How to test
1 - Check into current branch
2 - Do `npm install` and `npm run start:local` to run this PR locally
3 - Log as Owner user
4 - Go to dashboard → Other Links → Permissions Management.
5 - Test all the functionalities and see if everything works.

## New design
image

image

### CODE REVIEW

**Review by AuroraLovesWhales — DISMISSED** [review]
Hi Victor! I have tested out your new design. It looks great and the functionalities work the same. Great work!

Before:
image
image

After:
image
image

**Comment by LucileTech** [comment]
Hi @vitorsaa2k I tested your PR: 

Before
Screenshot 2023-04-20 at 14 47 41
Screenshot 2023-04-20 at 14 47 53

After

Screenshot 2023-04-20 at 14 48 03
Screenshot 2023-04-20 at 14 49 21


This is maybe not related to your PR, but for example, when I change the user permission it has no consequences : 
Here I change the permission on the badges:
Screenshot 2023-04-20 at 14 49 21

And when I am connected with this user the badges permission are still the same:

Screenshot 2023-04-20 at 14 51 19

**Comment by one-community** [comment]
Didn't ask for this but it is looking good, thank you. At least 2 more approvals and I'll move it to final review. Please address the merge conflicts too.

File: src/components/PermissionsManagement/PermissionsManagement.css [hunk 1]
```diff
@@ -3,19 +3,22 @@
...
   display: flex;
-  flex-direction: column;
-  row-gap: 1rem;
   align-items: center;
+  gap: 30px;
 }
 
 .permissions-management__title {
   font-size: 1.8rem;
+  margin-bottom: ;
```
Comment (abdel-lall) [inline src/components/PermissionsManagement/PermissionsManagement.css:21]: Missing value.

**Review by LucileTech — APPROVED** [review]
Hi @vitorsaa2k I tested your PR and it seems like it's working fine! Considering the bug mentioned earlier is not part of this pr

PR 3 vitor redesign permissions page and user permission component #783
PR 3 vitor redesign permissions page and user permission component #783 approved

**Review by AuroraLovesWhales — APPROVED** [review]

**Comment by EdwinLauM** [comment]
Looks great! Good job! 

Testing PR #783

Testing PR #783_2

**Comment by AriaYu927** [comment]
Hi @vitorsaa2k , I've tested your PR and everything works as expected.

**Comment by wang9hu** [comment]
Good job @vitorsaa2k ! I tested your code and everything looks great! I will skip the screenshots since they are the same as others.

**Review by AriaYu927 — APPROVED** [review]

**Comment by one-community** [comment]
Thank you everyone, moving this to final review

**Review by EvianTan — APPROVED** [review]

**Pull request merged**